### PR TITLE
[#6]added definitions and declaration for class Options

### PR DESCRIPTION
**Pull request by tamerharsh**
This class is having definitions and declaration for class Options viz; long and short command followed by their respective descriptions.

### CODE REVIEW

File: commandlineparser/include/Options.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#include<iostream>
```
Comment (inbangsa) [inline commandlineparser/include/Options.h]: I would prefer a space  ```#include  <iostream>```

Comment (inbangsa) [inline commandlineparser/include/Options.h]: Add a  Guard to protect this file from Multiple inclusion 

#ifndef OPTIONS_H
#define OPTIONS_H




#endif OPTIONS_H

File: commandlineparser/include/Options.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#include<string>
```
Comment (inbangsa) [inline commandlineparser/include/Options.h]: Add this class inside a namespace

File: commandlineparser/include/Options.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+//stores value of various command options 
+class Options
+{
+	private:
+	//different parameters of the options 
+	std::string short_command, long_command, short_description, long_description; 
+	
+	public:
+	//for only single long command without other arguments etc.like only port
+	Options(std::string long_command);
```
Comment (inbangsa) [inline commandlineparser/include/Options.h]: parameter name can be command,  will help user to pass short as well as long.. May be use can use - and -- to distinguish them respectively

File: commandlineparser/include/Options.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	// options with parameters
+	Options(std::string short_command, std::string long_command, std::string short_description, std::string  long_description);
+	
+	//setters for setting values
+	void  set_option_short_command(std::string short_command);
+	void  set_option_long_command(std::string long_command);
+	//std::string set_option_short_description( std::string short_description);
+	//std::string set_option_long_description(std::string  long_description);
+
```
Comment (inbangsa) [inline commandlineparser/include/Options.h]: Delete dead code.

File: commandlineparser/src/Options.cpp [hunk 5]
```diff
@@ -1 +1,25 @@
-#include<iostream>
\ No newline at end of file
+#include"Options.h"
```
Comment (inbangsa) [inline commandlineparser/src/Options.cpp]: Add space ```#include "options.h"```

File: commandlineparser/src/Options.cpp [hunk 6]
```diff
@@ -1 +1,25 @@
...
+	 set_option_short_command(short_command);
+	 set_option_long_command(long_command);
+ }
+ 
+ Options::Options(std::string long_command):short_description(""),short_command(""),long_description("")
+ {  set_option_long_command(long_command);
+ 
+ }
+ 
+void Options::set_option_short_command(std::string short_command){if(!short_command.empty()){this->short_command='-'+short_command;}}
```
Comment (inbangsa) [inline commandlineparser/src/Options.cpp]: follow one indentation type. Here you have used { in front of function other places its in line next to the function name

Comment (inbangsa) [reply commandlineparser/src/Options.cpp]: is this empty check really needed ?

**Comment by inbangsa** [comment]
#6 was about giving user option to set the parameters. this class is internal to the lib. We still need the interface for user to add options.

File: commandlineparser/include/Options.h (hunk 1 above)
Comment (tamerharsh) [reply commandlineparser/include/Options.h]: ok

Comment (tamerharsh) [reply commandlineparser/include/Options.h]: ok

File: commandlineparser/include/Options.h (hunk 3 above)
Comment (tamerharsh) [reply commandlineparser/include/Options.h]: As discussed we are going to drop this  facility of explicitly one parameter  in the options.

File: commandlineparser/include/Options.h (hunk 4 above)
Comment (tamerharsh) [reply commandlineparser/include/Options.h]: ok

File: commandlineparser/include/Options.h [hunk 7]
```diff
@@ -1,29 +1,32 @@
-#include<iostream>
-#include<string>
+#include <iostream>
+#include <string>
+#ifndef OPTIONS_H
```
Comment (inbangsa) [inline commandlineparser/include/Options.h:1]: Guards needs to be at the top of the file

File: commandlineparser/include/Options.h [hunk 8]
```diff
@@ -1,29 +1,32 @@
...
+#define OPTIONS_H
 
-//stores value of various command options 
-class Options
+//This class stores the values of various command options 
+namespace opts
```
Comment (inbangsa) [inline commandlineparser/include/Options.h]: use only one namespace for complete project.

File: commandlineparser/include/Options.h [hunk 9]
```diff
@@ -1,29 +1,32 @@
...
 {
+	class Options
+	{
 	private:
-	//different parameters of the options 
-	std::string short_command, long_command, short_description, long_description; 
-	
+		//different parameters of the options
```
Comment (inbangsa) [inline commandlineparser/include/Options.h:12]: why this extra spaces ?

File: commandlineparser/src/Options.cpp [hunk 10]
```diff
@@ -1,25 +1,42 @@
...
- Options::Options(std::string long_command):short_description(""),short_command(""),long_description("")
- {  set_option_long_command(long_command);
- 
- }
- 
-void Options::set_option_short_command(std::string short_command){if(!short_command.empty()){this->short_command='-'+short_command;}}
-void Options::set_option_long_command(std::string long_command){if(!long_command.empty()){this->long_command="--"+long_command;}}
+void opts::Options::set_option_short_command(std::string short_command)
+{
+	if(!short_command.empty()){this->short_command='-'+short_command;}
```
Comment (inbangsa) [inline commandlineparser/src/Options.cpp]: is this empty check really needed ? same comment was there in last review.

**Review by inbangsa — CHANGES_REQUESTED** [review]

File: commandlineparser/include/Options.h (hunk 7 above)
Comment (tamerharsh) [reply commandlineparser/include/Options.h:1]: done .

Comment (inbangsa) [reply commandlineparser/include/Options.h:1]: Where is the commit for this ?

File: commandlineparser/src/Options.cpp (hunk 10 above)
Comment (tamerharsh) [reply commandlineparser/src/Options.cpp]: Empty check is not required.

**Pull request merged**